common/persistence/cassandra/cassandraPersistence.go

```diff
@@ -1684,12 +1684,16 @@ func (d *cassandraPersistence) GetCurrentExecution(
 	currentRunID := result["current_run_id"].(gocql.UUID).String()
 	executionInfo := createWorkflowExecutionInfo(result["execution"].(map[string]interface{}))
 	replicationState := createReplicationState(result["replication_state"].(map[string]interface{}))
+	lastWriteVersion := common.EmptyVersion
+	if replicationState != nil {
```

Comment:
was it not supposed to be nil at all times?

Reply:
With the current logic, we still write into replication state